dataeng/jobs/analytics/SnowflakeSchemaBuilder.groovy

```diff
@@ -0,0 +1,45 @@
+package analytics
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_triggers
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_log_rotator
+import static org.edx.jenkins.dsl.AnalyticsConstants.common_publishers
+
+class SnowflakeSchemaBuilder {
+    public static def job = { dslFactory, allVars ->
+        dslFactory.job('snowflake-schema-builder') {
+            parameters {
+                stringParam('WAREHOUSE_TRANSFORMS_URL', allVars.get('WAREHOUSE_TRANSFORMS_URL'), 'URL for the Warehouse Transforms Repo.')
+                stringParam('WAREHOUSE_TRANSFORMS_BRANCH', allVars.get('WAREHOUSE_TRANSFORMS_BRANCH'), 'Branch of Warehouse Transforms to use.')
+                stringParam('SCHEMAS', allVars.get('SCHEMAS'), 'Snowflake schemas to work on')
+                stringParam('DBT_PROJECT', allVars.get('DBT_PROJECT'), 'dbt project in warehouse-transforms to work on.')
+                stringParam('DBT_PROFILE', allVars.get('DBT_PROFILE'), 'dbt profile from analytics-secure to work on.')
+                stringParam('DBT_TARGET', allVars.get('DBT_TARGET'), 'dbt target from analytics-secure to work on.')
+                stringParam('NOTIFY', allVars.get('NOTIFY','$PAGER_NOTIFY'), 'Space separated list of emails to send notifications to.')
+            }
+            logRotator common_log_rotator(allVars)
+            multiscm {
+                git {
+                    remote {
+                        url('$WAREHOUSE_TRANSFORMS_URL')
+                        branch('$WAREHOUSE_TRANSFORMS_BRANCH')
+                        credentials('1')
+                    }
+                }
+            }
+            triggers common_triggers(allVars)
+            wrappers {
+                timestamps()
+            }
+            publishers common_publishers(allVars)
+            steps {
+                virtualenv {
+                    pythonName('System-CPython-3.6')
```

Comment:
System-CPython-3.5.2 is what's available under Manage Jenkins -> Global Tool Configuration -> Python Installations

Does this actually work without one that exactly matches the name that you've given?

Reply:
I don't know, @doctoryes suggested this was available. It should work with either.

Reply:
Other jobs used that Python version? And closer to Python3.7, the better. But either version is fine with me.

Reply:
Other jobs that are defined for different jenkins instances maybe, but not any of ours.  We have to use the 3.5.2 version unless we install another one under Manage Jenkins -> Global Tool Configuration -> Python Installations.

Reply:
I understand that just doing the installation would make it harder to replicate the recreation of the Jenkins server.  But I wonder if it is more valuable to move ahead with 3.6 (or 3.7), since that's a direction we would be wanting to head in future anyway.

Reply:
We're probably all synced up offline about this, but just wanted to be clear that the plan currently is to add a newer python to the box and add the corresponding jenkins tool configuration, all while trying to keep the ansible in sync.

Reply:
I have a WIP PR for this that is going through the process now for the ansible side of things. I've manually installed 3.7 and configured Shining Panda through the UI to match what I expect those ansible changes will create in the future.